datafusion/sql/src/select.rs

```diff
@@ -778,9 +777,11 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         // Rewrite the HAVING expression to use the columns produced by the
         // aggregation.
         let having_expr_post_aggr = if let Some(having_expr) = having_expr_opt {
-            let having_expr_post_aggr =
-                rebase_expr(having_expr, &aggr_projection_exprs, input)?;
-
+            let having_expr_post_aggr = if is_constant_expression(having_expr) {
```

Comment:
The check needs to be for a false constant (not just a false constant) right? I'll suggest a test to show this